docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,298 @@
+_s3://instrument-bucket/instrument-name/experiment-id/_.
+1. EDP listens for events in
+_s3://instrument-bucket/instrument-name/experiment-id/*_.  After the
+specified duration or event count, a `package-objects-ready` event
+is generated and sent to EventBridge.
+1. A custom SNS topic is created for monitoring data package creation
+that Lab and Computational scientists subscribe to (`SNS_TOPIC_ARN`).
+1. A custom lambda fuction triggered by the `package-objects-ready`
+event processes the experiment files and generates a data package.
+Additional processing includes (but is not limited to):
+    - Decorating the data package with helpful additional files:
+        - `README.md`: Noting that the package was created by EDP,
+	a custom lambda function, and validated with a [Quilt
+	workflow](../advanced-features/workflows.md).
+        - [`quilt_summarize.json`](../Catalog/VisualizationDashboards.md#quilt_summarize.json)
+        - [`.quiltignore`](../advanced-features/.quiltignore.md)
+    - Package metadata creation and validation: Send an SNS
+    notification on [metadata
+    validation](../advanced-features/workflows.md) failure.
+
+    <!--pytest.mark.skip-->
+    ```python
+    import datetime
+    import functools
+    import os
+    import pathlib
+    import tempfile
+    import boto3
+    import quilt3 as quilt3
+    from aws_lambda_powertools import Logger
+
+    logger = Logger()
+    s3 = boto3.client("s3")
+    sns = boto3.client("sns")
+
+    # Configuration environment variables defined for Lambda function
+    WORKFLOW_NAME = os.environ.get("WORKFLOW_NAME") or ...
+    QUARANTINE_BUCKET_NAME = os.environ["QUARANTINE_BUCKET_NAME"]
+    SNS_TOPIC_ARN = os.environ["SNS_TOPIC_ARN"]
+    MAX_SNS_SUBJECT_LEN = 99
+    QUILT_URL = os.environ["QUILT_URL"]
+
+    # Empty quilt_summarize.json object
+    QUILT_SUMMARIZE_JSON_STR = "[]"
+
+    # README.md default Markdown
+    QUILT_README_STR = f"""#Quilt package auto-generated by EDP\n\n
+    Created on {datetime.date.today()} by an 
+    automated Lambda agent for the {WORKFLOW_NAME} workflow."""
+
+    # File system files for Quilt to ignore
+    QUILT_IGNORE_STR = """.DS_*
+    Icon
+    ._*
+    .TemporaryItems
+    .Trashes
+    .VolumeIcon.icns
+    """
+
+    # Define helpful additional data package files
+    beautify_files = {
+        "quilt_summarize.json": QUILT_SUMMARIZE_JSON_STR,
+        "README.md": QUILT_README_STR,
+        ".quiltignore": QUILT_IGNORE_STR,
+    }
+
+    @logger.inject_lambda_context
+    def lambda_handler(event, context):
+
+        # EDP event data
+        bucket = event["detail"]["bucket"]
+        prefix = event["detail"]["prefix"]
+    
+        # Add every file in the prefix folder to the new data package
+        pkg = quilt3.Package().set_dir(".", f"s3://{bucket}/{prefix}")
+    
+        # Decorate the data package with example required metadata (as defined by WORKFLOW_NAME)
+        meta = {
+            "Author": "EDP",
```

Comment:
Where did these values come from? "Made up" is fine.

Reply:
They are made up